### ISSUE TITLE
Track names with ±# to indicate octave auto-shifting is broken with Solo & Local Orchestra mode

### ISSUE THREAD

**Issue by Meowchestra** [parulina/bardmusicplayer#23]
BardMusicPlayer compiled as of commit 012f81c with PR 19

If you have a track name labeled with ``Instrument±Octave`` (such as ``Piano+1``), BardMusicPlayer shifts / sets octaves multiple times on initial loading; per the name specify & shifting window count. The octave shift count seems to unintentionally be getting set in the shift box when loading the track name, causing playback to be further out of range than intended. This affects both Solo & Local Orchestra mode.

**Recording:** https://streamable.com/2aokuv

I also noticed I cannot move the shift count back down to 0 in Local Orchestra mode if loading a name-shifted track, but this visual bug might be related to https://github.com/parulina/bardmusicplayer/issues/22.

**Recording:** https://streamable.com/thjqii

Edit: Caused by https://github.com/parulina/bardmusicplayer/commit/19337cc21225daec1a3a024114e0e34203d3e0f5?

**Comment by Meowchestra** [comment]
Reverting the changes made here seems to fix the problem.
https://github.com/parulina/bardmusicplayer/commit/19337cc21225daec1a3a024114e0e34203d3e0f5

**Comment by Meowchestra** [comment]
@parulina This also affects the 1.23 branch if you are planning a v1.23 release before the larger v1.4 memory overhaul.

**Comment by parulina** [comment]
Yeah that's incorrect, octave shift is supposed to apply to a temporary internal variable so as to not disturb the octave shift UI. Good catch, will fix.

**Comment by Meowchestra** [comment]
Unfortunately still not fixed with https://github.com/parulina/bardmusicplayer/commit/458220926688ed082fc0a56efae9abf197964836. Solo still double shifts on midi load (by name and octave box) and local orchestra is still buggy when shifting the octave count, skipping octaves.

**Comment by Meowchestra** [comment]
@parulina 

https://github.com/parulina/bardmusicplayer/commit/458220926688ed082fc0a56efae9abf197964836#diff-643496786450b76cb70bbdfb8651805dR236

This is causing the double visual shifting. Reverting this line change fixes the Local Orchestra issue.

https://github.com/parulina/bardmusicplayer/blob/1.23/FFBardMusicPlayer/Controls/BmpLocalPerformer.cs#L235

Basically just changing

```
int po = OctaveNum + bmpSeq.GetTrackPreferredOctaveShift(track);
```
to 
```
int po = bmpSeq.GetTrackPreferredOctaveShift(track);
```

---

https://github.com/parulina/bardmusicplayer/commit/19337cc21225daec1a3a024114e0e34203d3e0f5#diff-bd77cc8dd2cb827cc1dbc7f57e416b74R227

This line change specifically is causing the double octave shift for solo, from the name and octave box count being set on file load.

https://github.com/parulina/bardmusicplayer/blob/1.23/FFBardMusicPlayer/Controls/BmpPlayer.cs#L226-227

Basically just changing

```
            // this will also update the keyboard
            OctaveShift = player.GetTrackPreferredOctaveShift(player.LoadedTrack);
```
to
```
			UpdateKeyboard(player.LoadedTrack);
```
---

**With these two changes on the latest 1.23 branch source, visual keyboard shifting and in-game playback is perfect for both solo and local orchestra.** There might be a more proper fix, but this hotfixes the problem for me.

**Comment by trotlinebeercan** [comment]
thanks for looking into this. i'll have to wait until patch drops before i can test.

**Comment by Meowchestra** [comment]
Confirmed fixed with https://github.com/parulina/bardmusicplayer/pull/35/commits/5c66a20b6f164cd7419c4bb5a00645ad6cc4c06d. All issues seem fixed so closing this issue! Changes and fixes will need to be merged into the main branch once 1.23 is released.
